packages/connections/src/XHRConnection.js

```diff
@@ -0,0 +1,206 @@
+import AbstractConnection from "./AbstractConnection";
+import ConnectionEvent from "./ConnectionEvent";
+
+const ALLOWED_METHODS = ["GET", "POST", "PUT", "DELETE"];
+const ALLOWED_RESPONSE_TYPES = [
+  "arraybuffer",
+  "blob",
+  "document",
+  "json",
+  "text"
+];
+
+/**
+ * Basic XHR Connection
+ * @todo add/remove headers
+ */
+export default class XHRConnection extends AbstractConnection {
+  /**
+   * Initializes an Instance of XHRConnection
+   * @constructor
+   * @param {string} url – URL to be fetched
+   * @param {object} [options]
+   * @param {string} [options.method=GET] – used method
+   * @param {*} [options.body] – payload for request
+   * @param {object} [options.headers] – additional headers (like content-type)
+   * @param {string} [options.responseType]
+   */
+  constructor(url, options = {}) {
+    super(url);
+
+    const {
+      method = "GET",
+      body = null,
+      headers = {},
+      responseType = "json",
+      timeout = 0
+    } = options;
+
+    if (!ALLOWED_METHODS.includes(method)) {
+      throw new Error(
+        `Invalid method "${method}". Valid methods are: "${ALLOWED_METHODS.join('", "')}".`
+      );
+    }
+
+    if (!ALLOWED_RESPONSE_TYPES.includes(responseType)) {
+      throw new Error(
+        `Invalid response type "${responseType}". Valid reponse types are "${ALLOWED_RESPONSE_TYPES.join('", "')}".`
+      );
+    }
+
+    if (typeof headers !== "object") {
+      throw new Error(
+        `Invalid headers. Type must be "object", is "${typeof headers}".`
+      );
+    }
+
+    if (!Number.isSafeInteger(timeout) || timeout < 0) {
+      throw new Error(
+        `Invalid timeout. Type must 0 or positive, is "${timeout}".`
```

Comment:
`Invalid timeout. Must be an integer >= 0 and less than Number.MAX_SAFE_INTEGER, is "${timeout}".`